### PR TITLE
xarray accessor for cube and grid_mapping

### PR DESCRIPTION
**Pull request by forman**
**NOTE: this PR is based on PR #511**

Addresses both #112 and #391

A data's cube subset and its grid mapping can now be accessed through the `xcube` property of `xarray.Dataset` instances. This feature requires importing the `xcube.core.xarray`package. Let `dataset` be an instance of `xarray.Dataset`, then

* `dataset.xcube.cube` is a `xarray.Dataset` that contains all cube variables of `dataset`, namely the ones with dimensions `("time", [...,] y_dim_name, x_dim_name)`, where `y_dim_name`,  `x_dim_name` are determined by the dataset's grid mapping. May be empty, if `dataset` has no cube variables.
* `dataset.xcube.gm` is a `xcube.core.gridmapping.GridMapping` that describes the CF-compliant grid mapping of `dataset`. May be `None`, if `dataset` does not define a grid mapping.

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] Changes documented in `CHANGES.md`
* [ ] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

Remember to close associated issues after merge!

### CODE REVIEW

**Comment by forman** [comment]
FYI @AliceBalfanz , @dzelge , @pont-us

File: CHANGES.md [hunk 1]
```diff
@@ -61,6 +61,19 @@
...
+  instance of `xarray.Dataset`, then
+  - `dataset.xcube.cube` is a `xarray.Dataset` that contains all cube 
+     variables of `dataset`, namely the ones with dimensions 
+     `("time", [...,], y_dim_name, x_dim_name)`, where `y_dim_name`, 
+    `x_dim_name` are determined by the dataset's grid mapping.
+     May be empty, if `dataset` has no cube variables.
+  - `dataset.xcube.gm` is a `xcube.core.gridmapping.GridMapping` that 
+     describes the CF-compliant grid mapping of `dataset`. 
+     May be `None`, if `dataset` does not define a grid mapping.
+
```
Comment (TonioF) [inline CHANGES.md:76]: Why don't you include non_cube?

**Review by TonioF — APPROVED** [review]
Minor comments.
Regarding the old API stuff: I think it is unexpected that you can call methods on a dataset that do not even operate on that dataset,i.e., new and open. I would ditch these two. The others actually look like nice and convenient. One could think about creating a dedicated accessor for these, but if your main concern is maintenance, this is probably not the way to go.

File: CHANGES.md (hunk 1 above)
Comment (forman) [reply CHANGES.md:79]: Forgotten... added.

**Pull request merged**